setup.py

```diff
@@ -62,27 +77,115 @@ def package_vars(version_file):
+        "cryptography>=3.3.2",
+        "decorator>=4.4.2",
+        "distlib>=0.3.1",
+        "dnspython>=2.1.0",
+        "docutils>=0.15.2",
+        "ecdsa>=0.14.1",
+        "Faker>=5.8.0",
+        "filelock>=3.0.12",
+        "Flask>=1.1.2",
+        "Flask-Cors>=3.0.9",
+        "gunicorn>=20.0.4",
+        "identify>=1.5.5",
+        "idna>=2.10",
+        "iniconfig>=1.1.1",
+        "ipdb>=0.13.4",
+        "ipython>=7.18.1",
+        "ipython-genutils>=0.2.0",
+        "itsdangerous>=1.1.0",
+        "jedi>=0.17.2",
+        "Jinja2>=2.11.3",
+        "jmespath>=0.10.0",
+        "MarkupSafe>=1.1.1",
+        "marshmallow>=3.7.1",
+        "mccabe>=0.6.1",
+        "mongomock>=3.22.0",
+        "mypy-extensions>=0.4.3",
+        "nodeenv>=1.5.0",
+        "packaging>=20.9",
+        "paramiko>=2.7.2",
+        "parso>=0.7.1",
+        "pathspec>=0.8.0",
+        "pexpect>=4.8.0",
+        "pickleshare>=0.7.5",
+        "pluggy>=0.13.1",
+        "prompt-toolkit>=3.0.8",
+        "ptyprocess>=0.6.0",
+        "py>=1.10.0",
+        "pyasn1>=0.4.8",
+        "pycodestyle>=2.6.0",
+        "pycparser>=2.20",
+        "pyflakes>=2.2.0",
+        "Pygments>=2.7.4",
+        "pymongo>=3.11.1",
+        "PyNaCl>=1.4.0",
+        "pyparsing>=2.4.7",
+        "python-dateutil>=2.8.1",
+        "python-dotenv>=0.14.0",
+        "python-jose>=3.2.0",
+        "PyYAML>=5.4",
+        "regex>=2020.10.11",
+        "requests>=2.24.0",
+        "rsa>=4.7",
+        "s3transfer>=0.3.4",
+        "selenium>=3.141.0",
+        "sentinels>=1.0.0",
+        "six>=1.15.0",
+        "soupsieve>=2.1",
+        "sshtunnel>=0.3.1",
+        "text-unidecode>=1.3",
+        "toml>=0.10.1",
+        "traitlets>=5.0.5",
+        "TwoCaptcha>=0.0.1",
+        "typed-ast>=1.4.1",
+        "typing-extensions>=3.7.4.3",
+        "undetected-chromedriver>=2.1.2",
+        "urllib3>=1.25.11",
+        "validators>=0.18.2",
+        "wcwidth>=0.2.5",
+        "Werkzeug>=1.0.1",
+    ],
     extras_require={
         "test": [
-            "pre-commit",
-            # coveralls 1.11.0 added a service number for calls from
-            # GitHub Actions. This caused a regression which resulted in a 422
-            # response from the coveralls API with the message:
-            # Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
-            # 1.11.1 fixed this issue, but to ensure expected behavior we'll pin
-            # to never grab the regression version.
-            "coveralls != 1.11.0",
+            "black",
             "coverage",
-            "pytest-cov",
+            "flake8",
+            "moto",
+            "pre-commit",
             "pytest",
+            "pytest-cov",
+            "pytest-dockerc",
+            "pytest-dotenv",
+            "pytest-env",
+            "pytest-mock",
+            "pytest-pythonpath",
         ]
     },
-    entry_points={"console_scripts": ["eal = eal.example_aws_lambda:main"]},
```

Comment:
Is there some reason the (non-directory) files in the root of `src/` can't be moved into a `wsgi` directory (or call it whatever makes sense)?  That would more closely align with how the skeleton organizes Python libraries/modules and it would allow you to set up an `entry_point` like [this](https://github.com/cisagov/skeleton-python-library/blob/develop/setup.py#L106-L107).

Related, wherever possible, you should have useful (non-empty, non-stubbed) `__init__.py` and `__main__.py` files.

For a simple (non-skeleton) example to reference, take a look at [`cisagov/hash-http-content`](https://github.com/cisagov/hash-http-content).

Reply:
I can do that. But what is the purpose of the `src/` folder at that point? Wouldn't it be better to just change `src/` to `domain-manager-api/` and put all the files under that? That's the typical structure that you see in all python packages. https://pythonpackaging.info/02-Package-Structure.html

Reply:
The use of the `src/` directory has been a shift in the Python development community that is picking up momentum. The [official packaging guide for Python](https://packaging.python.org/tutorials/packaging-projects/) uses this directory layout now. You can see more discussion points in the [pull request](https://github.com/pypa/packaging.python.org/pull/859) for that change. Although old, [this blog post](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure) has some excellent points about why the `src/` structure is preferred to a flat structure. Major projects like the [cryptography](https://github.com/pyca/cryptography) package and organizations like NASA are [using this structure](https://github.com/nasa/fprime/tree/devel/Gds).

The long and the short of it is this is now the preferred way to structure Python packages and it has technical merit, so we use that structure.

Reply:
@mcdonnnj That is good to know. I've never knew that Python had an official guide for that. I've never found anything for it and in the pass I've only ever seen not using the `src/` structure. 

So are we also expected to make all of our applications we write packageable? I thought that's the main purpose of include the `setup.py` is for the purposes of packaging and publishing to Python Package Index. I can definitely see the use case for it when I get to skeletonizing the domain-manager-cli project, but in this case, this is an application that is intended to ship as a docker container.

Reply:
Making it a package means you can leverage `pip` to install the project and any of its dependencies as one action. As an example I use `pip` to install [cisagov/hash-http-content](https://github.com/cisagov/hash-http-content) in [cisagov/vdp-scanner-docker](https://github.com/cisagov/vdp-scanner-docker) by including it as a project requirement [here](https://github.com/cisagov/vdp-scanner-docker/blob/e56858082cbd4883afdddf3edb6468b3e6a4b725/src/requirements.txt#L2). You can safely ignore the use of `Pipfile` and [pipenv](https://github.com/pypa/pipenv) in that project as I was experimenting with that as a dependency control mechanism.

We use `pip` to install our projects whether they are deployed in Docker containers or built into AWS AMIs whenever possible as it reduces the complexity of installation and maintenance. Being able to distribute them using pypi is just a side  effect of packaging a Python project.

Reply:
Ok, cool, I'll get onto all that, I've just never heard of a flask application being distributed with `pip` before. So that's the confusion I have with it. I like to know the "why" to a lot of stuff, so hence all my questions. I don't think I have anymore at this point.

Reply:
Although it is an old project that does not conform to our current development practices, we do exactly that with [cisagov/ncats-webd](https://github.com/cisagov/ncats-webd)!

Reply:
Thanks for that code reference. Looking at it and looks like it will be a great help. Thanks for the responses and enlightening me on a few things. Definitely learned some stuff.

Reply:
@zenine07 thanks for the great questions, and @mcdonnnj thanks for the thorough answers!  I was tempted to pull "a dad" on Friday and say, "because that's the way it is!". 

Strong work all.  This project is developing a shine!  

💪 ✨

Reply:
@dav3r @mcdonnnj I restructured the project as requested in https://github.com/cisagov/domain-manager-api/pull/194/commits/ba1b4e05b9542313b3d5aefbcf238318ec886839. 

Let me know if this looks good to you now. If so, I'll rerequest reviews.

Reply:
Looks great now- thanks! 👍